app/models/post.rb

```diff
@@ -0,0 +1,20 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  after_save :user_posts_count
+  after_destroy :user_posts_count
+
+  # A method which returns the 5 most recent comments for a given post.
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  # A method that updates the posts count for a user.
+  def user_posts_count
+    author.update(posts_count: author.posts.count)
+  end
```

Comment:
- [ ] I was unable to create a post because there is no` posts_count` column in the schema and migration files; perhaps you meant `posts_counter`. Kindly correct this for proper execution of your project. ℹ️ 

<img width="757" alt="Untitled" src="https://github.com/lily4178993/nell-blog/assets/103386183/4f70255b-d279-4214-9bc8-86024ccde3ff">